frontend/src/database/PostCommentDatabase.tsx

```diff
@@ -0,0 +1,302 @@
+import PouchDB from 'pouchdb-browser';
+import find from 'pouchdb-find';
+import { Post } from './types/public/Post';
+import { PostDB } from './types/internal/PostDB';
+import { CommentDB } from './types/internal/CommentDb';
+import { FindResults } from './types/internal/FindResults';
+import { Comment } from './types/public/Comment';
+import { DbTypeMapper } from './DbTypeMapper';
+import { DbEntryMethaData } from './types/internal/DbEntryMethaData';
+import { AllDocumentsInterface } from './types/internal/AllDocumentsInterface';
+
+PouchDB.plugin(find);
+
+export class CommentDBWrapper {
+  private db: PouchDB.Database<{}>;
```

Comment:
What is the type parameter used for? If its the representation of the managed data structures, wouldn't it make sense to actually define that?

Reply:
I just copied it from the type defenitions. If you get the db object than it will has this type.